spec/services/admin_notices/dismiss_spec.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+RSpec.describe(AdminNotices::Dismiss) do
+  subject(:result) { described_class.call(id: admin_notice.id, guardian: current_user.guardian) }
+
+  let!(:admin_notice) { Fabricate(:admin_notice, identifier: "problem.test") }
+  let!(:problem_check) { Fabricate(:problem_check_tracker, identifier: "problem.test", blips: 3) }
+
+  context "when user is not allowed to perform the action" do
+    fab!(:current_user) { Fabricate(:user) }
+
+    it { is_expected.to fail_a_policy(:invalid_access) }
+  end
+
+  context "when user is allowed to perform the action" do
+    fab!(:current_user) { Fabricate(:admin) }
+
+    it { is_expected.to run_successfully }
+
+    it "sets the service result as successful" do
+      expect(result).to be_a_success
+    end
```

Comment:
I might be wrong but this is probably testing the same thing twice @Flink no?

Reply:
It certainly does sound like the second one is testing the implementation details of the first one. This is based on another spec, so I can make the change to both if this is redundant.